update-stable-ids/Jenkinsfile

```diff
@@ -0,0 +1,105 @@
+import groovy.json.JsonSlurper
+
+pipeline {
+	agent any
+	
+	stages {
+		stage('Check upstream builds succeeded'){
+			steps{
+				script{
+					def configStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/ConfirmReleaseConfigs/lastBuild/api/json"
+					def configStatusJson = new JsonSlurper().parseText(configStatusUrl.getContent())
+					if(configStatusJson['result'] != "SUCCESS"){
+						error("Most recent ConfirmReleaseConfigs build status: " + configStatusJson['result'] + ". Please complete a successful build.")
+					}
+				}	
+			}
+		}
+		stage('Setup: Rotate slice DBs'){
+			steps{
+				script{
+					dir('update-stable-ids'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "mysql -u$user -p$pass -e \'drop database if exists ${env.SLICE_PREVIOUS}; create database ${env.SLICE_PREVIOUS}\'"
+							sh "zcat  archive/${env.PREV_RELEASE_NUMBER}/${env.SLICE_TEST}_${env.PREV_RELEASE_NUMBER}_snapshot.dump.gz 2>&1 | mysql -u$user -p$pass ${env.SLICE_PREVIOUS}"
+							sh "mysqldump -u$user -p$pass ${env.SLICE_TEST} > ${env.SLICE_TEST}_${env.RELEASE_NUMBER}_snapshot.dump"
+							sh "gzip -f ${env.SLICE_TEST}_${env.RELEASE_NUMBER}_snapshot.dump"
+							sh "mysql -u$user -p$pass -e \'drop database if exists ${env.SLICE_CURRENT}; create database ${env.SLICE_CURRENT}\'"
+							sh "zcat  ${env.SLICE_TEST}_${env.RELEASE_NUMBER}_snapshot.dump.gz 2>&1 | mysql -u$user -p$pass ${env.SLICE_CURRENT}"
+						}
+					}
+				}
+			}
+		}
+		stage('Setup: Back up gk_central'){
+			steps{
+				script{
+					dir('update-stable-ids'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "mysqldump -u$user -p$pass -h${env.CURATOR_SERVER} ${env.GK_CENTRAL} > ${env.GK_CENTRAL}_${env.RELEASE_NUMBER}_before_st_id.dump"
+							sh "gzip -f ${env.GK_CENTRAL}_${env.RELEASE_NUMBER}_before_st_id.dump"
```

Comment:
Could a local variable for the dump file be used here so any changes to the name only need to be done in one spot?